networkx/algorithms/community/local.py

```diff
@@ -0,0 +1,126 @@
+"""
+Local Community Detection Algorithms
+
+Local Community Detection (LCD) aims to detected one or a few communities
+starting from certain source nodes in the network. This differs from Global
+Communty Detection (GCD), which aims to partition an entire network into
+communities.
+
+LCD is often useful when only a portion of the graph is known or the
+graph is large enough that GCD is infeasable
+
+[1]_ Gives a good introduction and overview of LCD
+
+References
+----------
+.. [1] Baltsou, Georgia, Konstantinos Christopoulos, and Konstantinos Tsichlas.
+   Local community detection: A survey. IEEE Access 10 (2022): 110701-110726.
+   https://doi.org/10.1109/ACCESS.2022.3213980
+
+
+"""
+
+__all__ = ["clauset"]
+
+
+def clauset(G, source, cutoff=None):
```

Comment:
Re: naming - what do we think about a function name that attempts to be descriptive as opposed to a proper noun? I.e. something like `greedy_source_expansion` instead of `clauset`?

Additionally, I'd vote to enforce keyword-only arguments in the signature. IMO, making `cutoff` keyword-only is uncontroversial. I don't feel as strongly about `source`. So - the question is, do we want users to be able to call this like `nx.clauset(G, n)` or force users to use `nx.clauset(G, source=n)`. I personally like option 2 ([explicit is better than implicit](https://peps.python.org/pep-0020/#the-zen-of-python)), but there are tons of functions in networkx that are called with a `(G, n)` pattern. Either way, I'll suggest the one that I'm mildly in favor of, but feel free to re-arrange the keyword-only-ness as you see fit!

```suggestion
def clauset(G, *, source, cutoff=None):
```

```

Reply:
RE: naming - I was unsure on this as well, I went back and forward on this a few times in my commits. In the early commits I called it `greedy_source_expansion` exactly as you suggested. I went for `clauset` in the end as there seemed to be several different greedy source expansion algorithms that each had pros and cons in different situations. [This](https://doi.org/10.1109/ACCESS.2022.3213980) touches on a number of the different algorithms. My thought was in future PR's I could add some of the other algorithms. Maybe `clauset_greedy_source_expansion` is a more descriptive name?